### ISSUE TITLE
Prefer USERPROFILE for Azure CLI Authorization Profile

### ISSUE THREAD

**Issue by mbrancato** [terraform-providers/terraform-provider-azurerm#1395]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7

### Affected Resource(s)

* azurerm_*

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "test" {
  name     = "test"
  location = "East US"
}
```

### Debug Output

```
2018/06/14 13:40:23 [ERROR] root: eval: *terraform.EvalConfigProvider, err: Azure CLI Authorization Profile was not found. Please ensure the Azure CLI is installed and then log-in with `az login`.
```

### Expected Behavior

Look for `.azure` folder in `$env:USERPROFILE` path

### Actual Behavior

Looks for `.azure` folder in `$env:HOMEDRIVE/$env:HOMEPATH` path

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

The azurerm provider is using another Go project call `go-homedir` which improperly prefers HOMEDRIVE/HOMEPATH over USERPROFILE. 

The HOMEDRIVE/HOMEPATH path is non-standard or at least deprecated. It is a relic of Windows NT4 and mostly relates to roaming profiles. As of Windows Vista, Microsoft has defined Known Folder Locations. Note that HOMEDRIVE, HOMEPATH, and HOMESHARE are not recognized system default environment variables.

I ran into this issue because my company laptop has the HOMEDRIVE and HOMEPATH environment variables set, and are different than USERPROFILE. I noticed others had also had this issue, referenced below.

### References

https://msdn.microsoft.com/en-us/library/dd378457(v=vs.85).aspx

https://github.com/terraform-providers/terraform-provider-azurerm/blob/c6fec095978bbc27d278db17074622b2c3550160/vendor/github.com/mitchellh/go-homedir/homedir.go#L120-L137

#417

**Comment by mbrancato** [comment]
Pending feedback and comments, I can submit a PR to fix.

**Comment by katbyte** [comment]
Hi @mbrancato,

Sorry to hear that you are having this problem. As this sounds like it would be an issue for anyone using this package it might be best to open an [issue there?](https://github.com/mitchellh/go-homedir)

**Comment by mbrancato** [comment]
@katbyte I was unsure if the inclusion of it as an upstream dependency would be updated in future binary builds. Many terraform modules have this issue. I will open the issue there, should we leave this open to update the dependency in this repo since it is not a submodule?

**Comment by katbyte** [comment]
Yes and please link to it/back here. Thanks!

**Comment by mbrancato** [comment]
Hey @katbyte  and @tombuildsstuff  - the upstream fix was merged, do you see any issues there or on this end? will the build process for the next release update the repo?

**Comment by tombuildsstuff** [comment]
@mbrancato sure we'll update the vendor to include this fix :)

**Comment by tombuildsstuff** [comment]
I've just opened https://github.com/terraform-providers/terraform-provider-azurerm/pull/1718 which includes the updated version of github.com/mitchellh/go-homedir which will fix this issue :)

**Comment by tombuildsstuff** [comment]
hey @mbrancato 

Just to let you know that we've just shipped support for this as a part of v1.13.0 of the AzureRM Provider :)

Thanks!

**Comment by mbrancato** [comment]
Thanks for the update @tombuildsstuff. I'm going to work in tests for this in our next deployments.

**Comment by irmbrady** [comment]
@tombuildsstuff I am using v1.13.0, but I am getting `No valid (unexpired) Azure CLI Auth Tokens found. Please run az login.` when I run a `terraform plan`.

However, running `az group list` (to test authentication) prints a list of resource groups from our Azure account.

Should v1.13.0 work without me having to run `az login` so frequently?

And just to confirm, when I do an `az login`, my `terraform plan` does work as expected, although the token seems to time out after a short while.

**Comment by tombuildsstuff** [comment]
@irmbrady 

> Should v1.13.0 work without me having to run az login so frequently?

Unfortunately not - this issue fixed a bug where the Azure CLI path wasn't loaded from a users home directory if their home directory was on a network share.

There's a separate issue tracking that in #502 (with a proposed solution in #1752) - essentially the issue is that we don't persist the refreshed Access Token from the Azure CLI since we don't own that file (and we were trying to not potentially cause issues with the Azure CLI by writing to a file that we don't own, for instance should the schema change). There's more details in #502 and we _probably_ have a path forward for it - but there's some larger changes in #1746 which needs to be completed first - I'd recommend subscribing to #502 for updates.

Thanks!

**Comment by irmbrady** [comment]
@tombuildsstuff Thanks for clarifying, and pointing me to #502 !
